### ISSUE TITLE
results.show() does not work

### ISSUE THREAD

**Issue by kiristern** [ultralytics/yolov5#9819]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### Bug

I created a new conda env and downloaded the requirements as per the instructions; then tried running the tutorial as is but cannot display the results with `results.show()`

I get the following message:
```
Using cache found in /home/azureuser/.cache/torch/hub/ultralytics_yolov5_master
YOLOv5 🚀 2022-10-16 Python-3.10.6 torch-1.12.1+cu102 CUDA:0 (Tesla V100-PCIE-16GB, 16160MiB)

Fusing layers... 
YOLOv5s summary: 213 layers, 7225885 parameters, 0 gradients
Adding AutoShape... 
image 1/2: 720x1280 2 persons, 2 ties
image 2/2: 1080x810 4 persons, 1 bus
Speed: 11.5ms pre-process, 4.0ms inference, 1.3ms NMS per image at shape (2, 3, 640, 640)

Error: no "view" rule for type "image/png" passed its test case
       (for more information, add "--debug=1" on the command line)
Error: no "view" rule for type "image/png" passed its test case
       (for more information, add "--debug=1" on the command line)
```

### Environment

YOLOv5 🚀 2022-10-16 Python-3.10.6 torch-1.12.1+cu102 CUDA:0 (Tesla V100-PCIE-16GB, 16160MiB)

### Minimal Reproducible Example

following https://github.com/ultralytics/yolov5/issues/36#issue-636702522 exactly.

### Additional

interestingly, the images save in the `runs/detect/exp` folder, but i'd like to be able to display the results. I am working in a jupyter notebook.... been trying to find a work around for hours:(

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by MartinPedersenpp** [comment]
It looks like an issue with PIL(LOW) because the results.show() uses a PIL image.show function.

You can add these lines to your script:
```
from matplotlib import pyplot as plt
plt.imshow(np.squeeze(results.render()))
plt.title("Yolo results")
plt.show()
```
Be aware that the render function will draw on your image, so if you want to do something with the image afterwards, you need to store a copy somewhere.

**Comment by glenn-jocher** [comment]
@kiristern works for me in PyCharm. I don't think remote notebooks will be able to show popup windows though.

<img width="1470" alt="Screenshot 2022-10-17 at 12 09 46" src="https://user-images.githubusercontent.com/26833433/196151559-1a706b08-68bd-4e09-abf1-0c96c8366a5c.png">

**Comment by glenn-jocher** [comment]
@kiristern @MartinPedersenpp I've opened PR #9825 to add notebook support for results.show()

<img width="748" alt="Screenshot 2022-10-17 at 13 02 01" src="https://user-images.githubusercontent.com/26833433/196161400-1c95ecb8-472d-413e-949a-6d15a74a411d.png">

**Comment by glenn-jocher** [comment]
Unfortunately check_imshow() in PR seems to crash CI. Not sure what the most reliable way to determine if imshow is available in that case.

**Comment by MartinPedersenpp** [comment]
@glenn-jocher Not sure if this helps, but it looks like the cv.imshow method in general does not work on jupyter notebooks, so if you can add a detector to the check_imshow(), you could switch to pyplot instead which should work.
I found this post regarding detecting if a script is running on a notebook: https://discourse.jupyter.org/t/find-out-if-my-code-runs-inside-a-notebook-or-jupyter-lab/6935/13

**Comment by glenn-jocher** [comment]
@MartinPedersenpp right, building a solution based off of a new is_notebook() function.

**Comment by glenn-jocher** [comment]
@MartinPedersenpp PR works now with is_notebook() and IPython.display.display, but it seems very slow. About 10 seconds to show both images:

```python
import torch

# PyTorch Hub Model
model = torch.hub.load('ultralytics/yolov5:results/display', 'yolov5s')  # or yolov5n - yolov5x6, custom

# Images
img = ['https://ultralytics.com/images/zidane.jpg', 'https://ultralytics.com/images/bus.jpg']  # or file, Path, PIL, OpenCV, numpy, list

# Inference
results = model(img)

# Results
results.print()  # or .show(), .save(), .crop(), .pandas(), etc.
results.show()
```

**Comment by glenn-jocher** [comment]
@kiristern good news 😃! Your original issue may now be fixed ✅ in PR #9825. 

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!

**Comment by MartinPedersenpp** [comment]
@glenn-jocher Glad to be of help 🙂
I must say, that I am learning so much from browsing the issues section of this repo and trying to help where I can.
I admire the work that you put into this repo and that you help people out every day.

**Comment by glenn-jocher** [comment]
@MartinPedersenpp awesome! I basically just try to make daily small incremental changes that improve the most aggregate UI/UX painpoints among the most users/use cases. It adds up over time :)

**Comment by kiristern** [comment]
Thank you so much for your prompt resolution, it works now!

**Comment by glenn-jocher** [comment]
You're welcome, @kiristern! I'm glad to hear that the issue is resolved for you. If you have any further questions or run into any more issues, feel free to reach out. Happy coding with YOLOv5!
